WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/ActivityLogListFragment.kt

```diff
@@ -87,12 +89,14 @@ class ActivityLogListFragment : Fragment(R.layout.activity_log_list_fragment) {
                 }
             }
 
-            val site = if (savedInstanceState == null) {
-                val nonNullIntent = checkNotNull(nonNullActivity.intent)
-                nonNullIntent.getSerializableExtra(WordPress.SITE) as SiteModel
-            } else {
-                savedInstanceState.getSerializable(WordPress.SITE) as SiteModel
-            }
+            val site: SiteModel = requireNotNull(
+                if (savedInstanceState == null) {
+                    val nonNullIntent = checkNotNull(nonNullActivity.intent)
+                    nonNullIntent.getSerializableExtraCompat(WordPress.SITE)
+                } else {
+                    savedInstanceState.getSerializableCompat(WordPress.SITE)
+                }
+            )
```

Comment:
Why do you think the updated style on your example is better?
- The current `getSite()` has one `SiteModel`, updated style has two `<SiteModel>`s. So I chose the current one since it's shorter with one type declaration.
- Updated style has two nested scopes, more than the current style. I put `requireNotNull` to the `return requireNotNull(site)` line to reduce the nested scope for readability.

Reply:
**DISCLAIMER**: This is highly subjective on my part so please take it with a grain of salt and only consider applying this minor (🔍) comment of mine, else please feel free to ignore this and any other such minor (🔍) comment coming from me.

> Why do you think the updated style on your example is better?

I'll try to answer each bullet-point separately.

> The current getSite() has one SiteModel, updated style has two <SiteModel>s. So I chose the current one since it's shorter with one type declaration.

I suggested this because I think it is better to have an explicit `getSerializableExtraCompat<SiteModel>(...)` instead of an implicit `getSerializableExtraCompat(...)` type reference on those, only because of that.

It is indeed shorter with one type declaration, but in this case I chose being explicit instead of shorter.

> Updated style has two nested scopes, more than the current style. I put requireNotNull to the return requireNotNull(site) line to reduce the nested scope for readability.

I suggested this because I think it is better to have the `requireNotNull(...)` as soon as possible and avoid creating a nullable `site` in the first place, just so that we get forced to do a `requireNotNull(site)` during return.

Imagine this code gets updated in the future, wouldn't the author prefer to work with a non-null `site`, that is after the variable is defined, instead of working with a nullable, only to call the `requireNotNull(site)` on return. 

It is indeed an extra nesting scope, but in this case I chose working with non-null instead of one less nesting scope.

Wdyt? 🤔

Reply:
Thanks for the explanation. It makes sense.
Done: https://github.com/wordpress-mobile/WordPress-Android/pull/18061/commits/62375b25b5aa531d01075bca43101c913fa708ee

Reply:
You're awesome! 🙇 ❤️ 🚀